odoo/addons/base/models/ir_mail_server.py

```diff
@@ -127,6 +127,24 @@ def _check_smtp_ssl_files(self):
             elif mail_server.smtp_ssl_private_key and not mail_server.smtp_ssl_certificate:
                 raise UserError(_('SSL certificate is missing for %s.', mail_server.name))
 
+    def write(self, vals):
+        if not vals.get('active', True):
+            active_usages = set()
+            for record in self:
+                active_usages.update(record._active_usages_compute())
```

Comment:
```suggestion
            active_usages = {*record._active_usages_compute() for record in self}
```

Reply:
I see the idea but it seems not allowed. I have changed it in one line like this:
active_usages = {usage for record in self for usage in record._active_usages_compute()}